src/analysis/classic/mage/arcane/normalizers/ArcaneMissiles.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  AnyEvent,
+  BeginChannelEvent,
+  CastEvent,
+  DamageEvent,
+  EndChannelEvent,
+  EventType,
+} from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import SPELLS from 'common/SPELLS/classic/mage';
+
+/**
+ * Arcane Missiles registers the 'tick' as a cast in logs but not the channeled spell.
+ * Add a beginchannel event for the first 'tick' and change the id.
+ * Subsequent ids are treated as 'ticks'.
+ */
+
+const NEW_ARCANE_MISSILES_TIMEOUT = 5000;
+const ARCANE_MISSILES_BOLT_COUNT = 5;
+
+export interface ArcaneMissilesLog {
+  arcaneMissilesCastEvents: CastEvent[];
+  arcaneMissilesDamageEvents: DamageEvent[];
+}
+
+export interface ArcaneMissilesBeginChannelEvent extends BeginChannelEvent {
+  arcaneMissilesCastEvents: CastEvent[];
+  arcaneMissilesDamageEvents: DamageEvent[];
+}
+
+function isArcaneMissilesEvent(event: AnyEvent) {
+  return (
+    (event.type === EventType.Cast || event.type === EventType.Damage) &&
+    event.ability?.guid === SPELLS.ARCANE_MISSILES.id
+  );
+}
+
+function createBeginChannelEvent(event: CastEvent): ArcaneMissilesBeginChannelEvent {
+  const beginChannel: ArcaneMissilesBeginChannelEvent = {
+    ...event,
+    type: EventType.BeginChannel,
+    isCancelled: true,
+    trigger: event,
+    __fabricated: true,
+    arcaneMissilesCastEvents: [],
+    arcaneMissilesDamageEvents: [],
+  };
+  event.ability.guid = SPELLS.ARCANE_MISSILES_CHANNELED.id;
```

Comment:
This doesn't look to do anything

Reply:
It changes the spell id of the 1st AM tick to the channeled spell id. This makes it possible to:
- Accurately count the Arcane Missile casts on the statistics page
- Display the channeled cast on the timeline
- Appropriately add the GCD for the cast which decreases downtime % displayed on the checklist

Reply:
Were you instead intending to change the id before creating the beginChannel const, so that it would be inserted into the beginChannel const and returned?

Reply:
On line 40, all the event info is added from the Arcane Missiles tick cast. On line 48, the id is updated to Arcane Missiles channeled cast. It's necessary to change it after the other info is gathered.

Reply:
But _why_ are you changing it at all? It's never used afterwards, you're immediately returning the beginChannel object and never doing anymore with the passed in event.

Reply:
It modifies the event which is used in other places. For example:

Without the change:
![image](https://user-images.githubusercontent.com/65823478/206929778-432959bf-ea52-4357-8117-34552eb5d71b.png)

With the change:
![image](https://user-images.githubusercontent.com/65823478/206929737-25a66b5f-351b-4e6f-b33f-795d86e3c2c7.png)

It does seem a little odd at the bottom, so I moved it to the top.

Reply:
Moving it to the top causes it to go into the fabricated event, which makes sense why it would have an impact further on since that is returned from the function, the previous thing shouldn't have had any impact since it was unused

Reply:
It did work, screenshots are from the id at the bottom but I moved it to the top just in case.

Reply:
@Pewtro - Everything good here to merge?

Reply:
Yeah